lib/src/types/binary_codec.dart

```diff
@@ -796,7 +796,15 @@ class PostgresBinaryDecoder {
         if (dinput.timeZone.forceDecodeDateAsUTC) {
           return DateTime.utc(2000).add(Duration(days: value));
         }
-        return DateTime(2000).add(Duration(days: value));
+        // https://github.com/dart-lang/sdk/issues/56312
+        // ignore past timestamp transitions and use only current timestamp in local datetime
+        final nowDt = DateTime.now();
+        var baseDt = DateTime(2000);
+        if (baseDt.timeZoneOffset != nowDt.timeZoneOffset) {
+          final difference = baseDt.timeZoneOffset - nowDt.timeZoneOffset;
```

Comment:
I'm wondering, without fully seeing the context, if this operation is always valid. E.g. can this be close to or larger than a full day? Or instead of substracting a half hour, are we adding 23.5 hours, just because of a strange combination?

Reply:
I believe that there will not be this problem because the timezone transitions of the past are more related to the daylight saving time of countries that by law no longer have daylight saving time.